charts/kashti/templates/ingress.yaml

```diff
@@ -15,15 +16,28 @@ metadata:
       {{ $key }}: {{ $value | quote }}
     {{- end }}
 spec:
+  {{- if .Values.ingress.ingressClassName }}
+  ingressClassName: {{ .Values.ingress.ingressClassName }}
+  {{- end }}
```

Comment:
Should this also be conditional on the v1 stable API being used?

Reply:
Actually... it seems `ingressClassName` is also in the beta API starting with 1.18.

So, I guess that's the condition.

Reply:
I added a check for the Kubernetes version, but also left part of the responsibility to the user. That's because the Kubernetes version check isn't sufficient. The ingress controller version also needs to support the field, e.g.: https://github.com/kubernetes/ingress-nginx/pull/5307

Reply:
Thanks! Totally agree that the user still bears a lot of responsibility for this.